src/mdns/mdns_mdnssd.cpp

```diff
@@ -435,12 +435,13 @@ void PublisherMDnsSd::HandleServiceRegisterResult(DNSServiceRef         aService
 
 void PublisherMDnsSd::DiscardService(const char *aName, const char *aType, DNSServiceRef aServiceRef)
 {
+    OT_UNUSED_VARIABLE(aServiceRef);
+
     ServiceIterator service = FindPublishedService(aName, aType);
 
     if (service != mServices.end())
     {
         assert(aServiceRef == nullptr || aServiceRef == service->mService);
-        OTBR_UNUSED_VARIABLE(aServiceRef);
```

Comment:
Is it better to update `OTBR_UNUSED_VARIABLE` to make it work with `clang`?

Reply:
I don't see a way to achieve that. Do you have any idea?

Reply:
OK, I see. The reason clang complains about it is that it's in an if statement? That's weird.